### ISSUE TITLE
Support for port forwarding from host to guest

### ISSUE THREAD

**Issue by ijc** [moby/vpnkit#257]
I'm working with [LinuxKit](https://github.com/linuxkit/linuxkit) and would like to be able to support port forwarding with the Hyperkit+Vpnkit backend on MacOS in the same way as the QEMU backend works.

With the QEMU backend I can do:
```
linuxkit run qemu -publish 1234:22 «image»
```
which maps `localhost:1234` to port `22` on the VMs IP address, with that I can `ssh -p 1234 localhost` to reach an ssh server running in the VM. This essentially uses the `hostfwd` argument which QEMU supports with the `-net` option.

Note that there is no need for any in VM process other than sshd itself since this does TCP forwarding to the VM over the Ethernet link, targeting the IP address which QEMU knows because it assigned it.

I'm trying to make
```
linuxkit run hyperkit -publish 1234:22 «image»
```
behave the same such that `linuxkit run -publish 1234:22 «image»` works regardless of the host OS (on Linux this will autodetect QEMU, on MacOS it will use hyperkit)

I began adding support using the `github.com/moby/vpnkit/go/pkg/vpnkit` library to talk to the vpnkit instance in order to ask it to perform the forwarding (using the [`vpnkit-expose-port`](https://github.com/moby/vpnkit/blob/master/go/cmd/vpnkit-expose-port/main.go) code for reference).

I ran into a roadblock however which is that I do not know the guest IP address in order to make the request. I'm also lead to believe that the current vpnkit forwarding arrangements require an agent with the VM which I'd like to avoid.

So I (think I) have two asks from VPNKit so I can build this:

* Provide a mechanism to retrieve the IP address dynamically assigned to the VM, perhaps based off the VPNKit UUID? Note that VPNKit supports static IP address allocation based on a magic syntax for the UUID (where the top 4 bytes encode the IP and the other bytes are zero). I want to avoid having to statically assign IP addresses though.
* Ensure that the port forwarding can work without the need for an in guest agent.

My incomplete code (which doesn't compile since I don't have a guest IP address to use) can be found [in my tree](https://github.com/ijc/linuxkit/tree/hyperkit-port-forwarding)

**Comment by ijc** [comment]
TL;DR I don't actually need this functionality myself any more, but I think it would be useful functionality, but with much lower priority.

My main reason for this was parity between the `linuxkit run` backends for hyperkit and qemu, another side of which was making `linuxkit run` with QEMU on Linux be able to do inter-VM networking (which the hyperkit+vpnkit backend already does seemlessly) (essentially I wanted [this script](https://github.com/linuxkit/linuxkit/blob/master/projects/kubernetes/boot.sh) to have no platform dependencies). It turns out the QEMU options for doing that are incompatible with port forwarding there so I've had to take another approach which doesn't involve port forwarding.
